File: elixir/grains/lib/grains.ex
defmodule Grains do
  def square(number) when number in 1..64 do
    {:ok, 2 ** (number - 1)}
  end
  def square(_number) do
    {:error, "The requested square must be between 1 and 64 (inclusive)"}
  end

  def total, do: {:ok, 2 ** 64 - 1}
end

File: c/grains/grains.c
#include "grains.h"

uint64_t square(uint8_t index)
{
    if (index < 1 || index > 64)
        return 0;

    return 1ul << (index - 1);
}

uint64_t total(void)
{
    return (1ul << 63ul << 1ul) - 1;
}
